### ISSUE TITLE
Dimensional Anomaly Floor Replacement lead to no plating under an airlock tile.

### ISSUE THREAD

**Issue by Bjarl** [tgstation/tgstation#70161]
## Round ID: [191260](https://scrubby.melonmesa.com/round/191260)

Server revision compiled on: 
Master commit: [0aa9aee51f6394dba715eb9b622ec09bfc25948d](https://www.github.com/tgstation/tgstation/commit/0aa9aee51f6394dba715eb9b622ec09bfc25948d)
The following pull requests are currently test merged:
[#69838: 'Micros the lighting subsystem (Saves a second of init)' by LemonInTheDark at commit 9f8144427a](https://www.github.com/tgstation/tgstation/pull/69838)
[#70124: 'Fixes gravity pulse and transparent floor plane sharing a layer' by Time-Green at commit 5097a3effc](https://www.github.com/tgstation/tgstation/pull/70124)
## Reproduction:
![g1lX1KYVC2](https://user-images.githubusercontent.com/94164348/192449114-2382548d-4287-487a-87d0-7049101b431f.png)

![dreamseeker_gizEmoooqa](https://user-images.githubusercontent.com/94164348/192448845-c6db726a-838f-40c4-a9b8-e1277bc91bda.png)
![dreamseeker_CdlLWUvRdM](https://user-images.githubusercontent.com/94164348/192448856-768b7b64-9edc-4a07-b23e-5f13587cc29c.png)
This is the art storage maint airlock, right next to the art storage room. It leads into cargo maint
Crowbar this tile after a dimensional anomaly turns it into plasma floor
Fall through the newly created hole to z-3 and die instantly from the plasma lake

![dreamseeker_36EvUQMujp](https://user-images.githubusercontent.com/94164348/192452249-581e3ac5-3de7-41d5-a3f4-b825960a6e2d.png)

I couldn't reproduce in round, but my uneducated guess says it's either something wrong with that tile in specific, or the dimensional anomaly replacing things under doors.

**Comment by Bjarl** [comment]
Pumpkin0 helped me do some testing, we discovered a pattern in that when it overrides a tile that's base plating (such as in maint), it goes to an openspace when crowbarred, but this behavior isn't consistent across all materials. We found that gold, lavaland tiles, and glass tiles have this behavior, we didn't test them all due to not being able to test specific themes of the anomaly. 
We also tested anomaly armor and found that it did not share this behavior with a few materials (glass, meat, snow).

TLDR
Anomaly overrides plating with nothing under it -> goes to space when crowbarred
Anomaly Armor doesn't do this but makes ghost tiles under it sometimes.

**Comment by msgerbs** [comment]
So for the ones that it doesn't do this for, what does it do for plating? Or I guess, what _should_ the behavior be in your mind? Should it create a new material floor tile if there's not already a tile there?

**Comment by Bjarl** [comment]
That's what I expected it to be doing

**Comment by msgerbs** [comment]
I think that's fair. I'm not actually sure what the _intended_ behavior is but I agree that seems like a sensible way to handle it.
